File: android/src/main/kotlin/com/github/hugochou/amap_search/FunctionRegistry.kt
package com.github.hugochou.amap_search

import com.github.hugochou.amap_search.search.*

/**
 * 搜索功能集合
 */
val SEARCH_METHOD_HANDLER: Map<String, SearchMethodHandler> = mapOf(
        "setApiKey" to SetApiKey,
        "updatePrivacyStatement" to UpdatePrivacyStatement,
        "calculateDriveRoute" to CalculateDriveRoute,
        "searchPoi" to SearchPoiKeyword,
        "searchPoiBound" to SearchPoiBound,
        "searchPoiPolygon" to SearchPoiPolygon,
        "searchPoiId" to SearchPoiId,
        "searchRoutePoiLine" to SearchRoutePoiLine,
        "searchRoutePoiPolygon" to SearchRoutePoiPolygon,
        "searchGeocode" to SearchGeocode,
        "searchReGeocode" to SearchReGeocode,
        "searchBusStation" to SearchBusStation,
        "distanceSearch" to DistanceSearchHandler
)

File: ios/Classes/FunctionRegistry.m
//
// Created by Yohom Bao on 2018-12-12.
//

#import "FunctionRegistry.h"
#import "IMethodHandler.h"
#import "SearchHandlers.h"
//#import "NaviHandlers.h"

static NSDictionary<NSString *, NSObject <SearchMethodHandler> *> *_searchDictionary;

@implementation SearchFunctionRegistry {

}
+ (NSDictionary<NSString *, NSObject <SearchMethodHandler> *> *)searchMethodHandler {
  if (!_searchDictionary) {
    _searchDictionary = @{
      @"setApiKey": [SetApiKey alloc],
      @"updatePrivacyStatement": [UpdatePrivacyStatement alloc],
      @"calculateDriveRoute": [CalculateDriveRoute alloc],
      @"searchPoi": [SearchPoiKeyword alloc],
      @"searchPoiBound": [SearchPoiBound alloc],
      @"searchPoiPolygon": [SearchPoiPolygon alloc],
      @"searchPoiId": [SearchPoiId alloc],
      @"searchRoutePoiLine": [SearchRoutePoiLine alloc],
      @"searchRoutePoiPolygon": [SearchRoutePoiPolygon alloc],
      @"searchGeocode": [SearchGeocode alloc],
      @"searchReGeocode": [SearchReGeocode alloc],
      @"distanceSearch":[DistanceSearch alloc],
      @"searchBusStation":[SearchBusStation alloc],
    };
  }
  return _searchDictionary;
}

@end
